src/app/cdk/side-bar/modals/modal-email/modal-email.component.ts

```diff
@@ -310,22 +331,47 @@ export class ModalEmailComponent implements OnInit, OnDestroy {
 
   saveEvent() {
     if (this.emailsForm.valid) {
-      const data = this.formToOtherNames(this.emailsForm, this.emails)
-      const emailNewPrimary = data.filter((email) => email.primary)[0].value
-      const emailOldPrimary = this.emails.filter((email) => email.primary)[0]
-        .value
+      const data = this.formToBackend(this.emailsForm, this.emailsActions)
+      if (this.emailsDelete.length > 0) {
+        this.emailsDelete.forEach((emailDelete) => {
+          this._recordEmails.delete(emailDelete.email).pipe(first()).subscribe()
+        })
+      }
+      if (this.emailPrimary.length > 0) {
+        this.emailPrimary.forEach((emailPrimary) => {
+          this._recordEmails
+            .setAsPrimaryEmail(emailPrimary.email)
+            .pipe(first())
+            .subscribe()
+        })
+      }
+      const emailNewPrimary = data.filter(
+        (emailActions) => emailActions.email.primary
+      )[0].email
+      const emailOldPrimary = this.emailsActions.filter(
+        (emailActions) => emailActions.email.primary
+      )[0].email
       const otherNames = {
-        emails: data.filter((email) => !email.primary),
+        emails: data
+          .filter((emailActions) => !emailActions.email.primary)
+          .map((emailActions) => emailActions.email),
         errors: [],
       }
-      if (emailNewPrimary !== emailOldPrimary) {
+      if (emailNewPrimary.value !== emailOldPrimary.value) {
         this._recordEmails
-          .editEmail(emailOldPrimary, emailNewPrimary)
+          .editEmail(emailOldPrimary.value, emailNewPrimary.value)
```

Comment:
With the new call to the `setAsPrimaryEmail` is it require to check if the oldPrimary and newPrimary are not equal to edit the old email? It seems like is not required anymore, can you confirm?

Reply:
Will the new logic work ok if I set a second email as primary, and also edit the previous primary email? 
Seems to me the second email will be called to be primary on the subscription line `345`
and at the same time, the old primary email will be called to be renamed as the new email primary email at the subscribe on line `364`

Reply:
I changed to approach to fix the problem with the emails and I noticed that there was an error in the backend that wasn't let change the primary email https://github.com/ORCID/ORCID-Source/pull/6337